client/analytics-service-rest-client/src/main/java/org/hawkular/apm/analytics/service/rest/client/AnalyticsServiceRESTClient.java

```diff
@@ -1121,80 +327,19 @@ protected boolean buildQueryString(StringBuilder builder, Criteria criteria) {
      */
     @Override
     public Collection<CommunicationSummaryStatistics> getCommunicationSummaryStatistics(String tenantId,
-            Criteria criteria, boolean tree) {
+                                                                                        Criteria criteria, boolean tree) {
         if (log.isLoggable(Level.FINEST)) {
             log.finest("Get communication summary statistics: tenantId=[" + tenantId + "] criteria="
                     + criteria + " as tree? " + tree);
         }
 
-        StringBuilder builder = new StringBuilder()
-                .append(getUri())
-                .append("hawkular/apm/analytics/communication/summary");
-
-        if (buildQueryString(builder, criteria)) {
-            builder.append('&');
+        if (criteria.parameters().isEmpty()) {
```

Comment:
As we are passing criteria in json form, we could get rid of this parameter encoding approach - so instead it may be better to have an 'isDefined()' method on the Criteria class to determine if any of the fields have been set?